ci-operator/step-registry/ipi/conf/gcp/user-labels-tags/ipi-conf-gcp-user-labels-tags-commands.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+PATCH="${SHARED_DIR}/user-labels-tags.yaml.patch"
+cat > "${PATCH}" << EOF
+platform:
+  gcp:
+    userLabels:
```

Comment:
If we are going to use `USER_TAGS` env to set userTags, I guess we also can do similar things for `userLabels`.

Reply:
The workflow aims to cover the basic scenario of userLabels & userTags testing, and the syntax testing of userLabels & userTags is beyond the scope. FYI I'm working on 2 regression test cases for such testing, see [OCP-67066](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-67066) and [OCP-67068](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-67068), both with Low priority.

Reply:
We had ever gotten escalated issues from OSD team to complain our basic userTags coverage is not enough on aws, that is why we added several complex userTags strong for that purpose.

Reply:
Per the definition of `userTags` and `userLables` [1], various character combinations should be covered in CI, and let's configure the maximum number of tags: 50.

[1]
```
KIND:     InstallConfig
VERSION:  v1

RESOURCE: <[]object>
  userLabels has additional keys and values that the installer will add as labels to all resources that it creates on GCP. Resources created by the cluster itself may not include these labels. This is a TechPreview feature and requires setting CustomNoUpgrade featureSet with GCPLabelsTags featureGate enabled or TechPreviewNoUpgrade featureSet to configure labels.

FIELDS:
    key <string> -required-
      key is the key part of the label. A label key can have a maximum of 63 characters and cannot be empty. Label must begin with a lowercase letter, and must contain only lowercase letters, numeric characters, and the following special characters `_-`.

    value <string> -required-
      value is the value part of the label. A label value can have a maximum of 63 characters and cannot be empty. Value must contain only lowercase letters, numeric characters, and the following special characters `_-`.

KIND:     InstallConfig
VERSION:  v1

RESOURCE: <[]object>
  userTags has additional keys and values that the installer will add as tags to all resources that it creates on GCP. Resources created by the cluster itself may not include these tags. Tag key and tag value should be the shortnames of the tag key and tag value resource. This is a TechPreview feature and requires setting CustomNoUpgrade featureSet with GCPLabelsTags featureGate enabled or TechPreviewNoUpgrade featureSet to configure tags.

FIELDS:
    key <string> -required-
      key is the key part of the tag. A tag key can have a maximum of 63 characters and cannot be empty. Tag key must begin and end with an alphanumeric character, and must contain only uppercase, lowercase alphanumeric characters, and the following special characters `._-`.

    parentID <string> -required-
      parentID is the ID of the hierarchical resource where the tags are defined, e.g. at the Organization or the Project level. To find the Organization ID or Project ID refer to the following pages: https://cloud.google.com/resource-manager/docs/creating-managing-organization#retrieving_your_organization_id, https://cloud.google.com/resource-manager/docs/creating-managing-projects#identifying_projects. An OrganizationID must consist of decimal numbers, and cannot have leading zeroes. A ProjectID must be 6 to 30 characters in length, can only contain lowercase letters, numbers, and hyphens, and must start with a letter, and cannot end with a hyphen.

    value <string> -required-
      value is the value part of the tag. A tag value can have a maximum of 63 characters and cannot be empty. Tag value must begin and end with an alphanumeric character, and must contain only uppercase, lowercase alphanumeric characters, and the following special characters `_-.@%=+:,*#&(){}[]` and spaces.
```